cmd/okctl/delete.go

```diff
@@ -125,13 +124,18 @@ including VPC, this is a highly destructive operation.`,
 
 			formatErr := o.ErrorFormatter(fmt.Sprintf("delete cluster %s", opts.Environment), userDir)
 
-			err = services.Domain.DeletePrimaryHostedZone(o.Ctx, client.DeletePrimaryHostedZoneOpts{
+			err = services.Domain.DeletePrimaryHostedZone(o.Ctx, o.CloudProvider, client.DeletePrimaryHostedZoneOpts{
```

Comment:
I should have caught this last time, but we should not be injecting the `CloudProvider` here, this is a detail of the concrete implementation of the service itself.

We should rather be injecting the `CloudProvider` when creating the `services.Domain`